docs/portingguide/Porting_Guide.md

```diff
@@ -0,0 +1,80 @@
+ - Focus on simplicity and correctness.
+ - Optimize the port later.
+
+In MMTk’s language, a plan is essentially a configuration which specifies a GC algorithm.   Plans can be selected at run time.   Not all plans will be suitable for all runtimes.   For example, a runtime that for some reason cannot support object movement won’t be able to use plans that use copying garbage collection.
+
+
+### NoGC
+We always start a port with NoGC.  It is the simplest possible plan, simply allocates memory and never collects it.   Although this appears trivial, depending on the complexity of the runtime and how well factored (or not) its internal GC interfaces are, just getting this working may be a major undertaking.   In the case of V8, the refactoring within V8 required to get a simple NoGC plan working was substantial, touching over 100 files.   So it’s a good idea not to underestimate the difficulty of a NoGC port!
+
+In order to implement NoGC, we only need to handle MMTk initialisation (`gc_init`), mutator initialisation (`bind_mutator`), and memory allocation (`alloc`).
+
+You may want to take the following steps.
+ 
+1. Set up the binding repository/directory structure:
+    - `/mmtk` - the MMTk side of the binding. To start, this can be an almost direct copy of the [Dummy VM binding](https://github.com/mmtk/mmtk-core/tree/master/vmbindings/dummyvm).
+    - `/vm` (rename this to your VM name) - VM-specific code for integrating with MMTk. This should act as a bridge between the generic GC interface offered by the VM and the MMTk side of the binding.
+    - `/repos` - clone your VM code here. Optionally also clone a copy of mmtk-core if you plan on modifying MMTk itself 
+    - You may also find it helpful to take inspiration from the OpenJDK binding, particularly for a more complete example of the relevant `Cargo.toml` files (note: the use of submodules is no longer recommended): https://github.com/mmtk/mmtk-openjdk 
+2. Change the VM build process to build and/or link MMTk
+    - It may be easier to simply build a static and/or dynamic binary for MMTk and link it to the language directly, manually building new binaries as necessary. 
+        1. `cd binding_repo/mmtk`
+        2. `cargo build --features nogc` to build in debug mode or add `--release` for release mode
+        3. Copy the shared or static library from `target/debug` or `target/release` to your desired location
+    - Later, you can edit the language build process to build MMTk at the same time automatically.
+3. Replace VM allocation with calloc
+    - Change all alloc calls in the GC to calloc (https://www.tutorialspoint.com/c_standard_library/c_function_calloc.htm). Note: calloc is used instead of malloc as it zero-initialises memory.
+    - The purpose of this step is simply to help you find all allocation calls.
+4. Single Threaded MMTk Allocation
+    1. Include the [`mmtk.h` header file](https://github.com/mmtk/mmtk-core/blob/master/vmbindings/dummyvm/api/mmtk.h)
```

Comment:
The binding implementer should have their own `mmtk.h` based on API functions exposed to native. `mmtk.h` provided in dummyvm is just an example.